agents/agent_similarity.py

```diff
@@ -42,7 +43,15 @@ def calculate_similarity_threshold(self) -> float:
         :return: 98th percentile of similarity threshold.
         """
         try:
-            embeddings = [self.get_embedding(agent.purpose) for agent in self.agents]
+            embeddings=[]
+            for agent in self.agents:
+                if agent.embedding_purpose is None:
+                   agent.embedding_purpose = self.get_embedding(agent.purpose)
+                else:
+                   print("\nUsing embedding_purpose variable instead of api lookup")
```

Comment:
I suggest to replace this with a debug output

Reply:
Happy new year! 
I just removed the print statements.  They were there mainly to show that we were looking up the embeddings more than once via the api, and that this could be eliminated by the additions I added.